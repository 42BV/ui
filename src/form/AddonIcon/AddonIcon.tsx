import { Icon, IconType } from '../../core/Icon';

import { InputGroupText } from 'reactstrap';

export type Props = {
  icon: IconType;

  /**
   * Optional extra CSS class you want to add to the component.
   * Useful for styling the component.
   */
  className?: string;
};

/**
 * Defines an addon to use with an Input element which is a readonly
 * Icon.
 */
export function AddonIcon({ icon, className }: Props) {
  return (
    <InputGroupText>
      <Icon icon={icon} className={className} />
    </InputGroupText>
  );
}
