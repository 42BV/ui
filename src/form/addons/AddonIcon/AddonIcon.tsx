import React from 'react';
import { Icon, IconType } from '../../..';

import { Addon, Props as AddonProps } from '../Addon/Addon';
import { InputGroupText } from 'reactstrap';

export type Props = Omit<AddonProps, 'children'> & {
  icon: IconType;

  /**
   * Optional extra CSS class you want to add to the component.
   * Useful for styling the component.
   */
  className?: string;
};

/**
 * Defines an addon to use with an Input element which is a readonly
 * Icon.
 */
export function AddonIcon({ icon, className, position }: Props) {
  return (
    <Addon position={position}>
      <InputGroupText>
        <Icon icon={icon} className={className} />
      </InputGroupText>
    </Addon>
  );
}
