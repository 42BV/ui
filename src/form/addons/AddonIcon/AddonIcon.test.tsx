import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { AddonIcon } from './AddonIcon';

describe('Component: AddonIcon', () => {
  function setup() {
    const addonIcon = shallow(
      <AddonIcon icon="360" position="right" className="custom-class" />
    );

    return { addonIcon };
  }

  test('ui', () => {
    const { addonIcon } = setup();

    expect(toJson(addonIcon)).toMatchSnapshot('Component: AddonIcon');
  });
});
