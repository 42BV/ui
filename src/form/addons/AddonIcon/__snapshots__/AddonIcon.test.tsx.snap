// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AddonIcon ui: Component: AddonIcon 1`] = `
<Addon
  position="right"
>
  <InputGroupText
    tag="span"
  >
    <Icon
      className="custom-class"
      icon="360"
    />
  </InputGroupText>
</Addon>
`;
